source/stack_alloc.c

```diff
@@ -36,7 +36,7 @@ void* _ngf_sa_alloc(_ngf_sa *allocator, size_t nbytes) {
   void           *result             = NULL;
   const ptrdiff_t consumed_capacity  = allocator->ptr - allocator->data;
   const ptrdiff_t available_capacity = allocator->capacity - consumed_capacity;
-  if (available_capacity >= nbytes) {
+  if (available_capacity >= (ptrdiff_t)nbytes) {
```

Comment:
another option is cast `available_capacity` to `size_t`, but imho it's not better )